examples/example.c

```diff
@@ -503,11 +500,50 @@ main(int argc, char **argv)
             options, "./sentry_crash_reporter");
 #endif
     }
+    if (has_arg(argc, argv, "log-attributes")) {
+        sentry_options_set_logs_with_attributes(options, true);
+    }
+
+    // sentry_options_set_sdk_name(options, "strange-sdk-name");
 
     if (0 != sentry_init(options)) {
         return EXIT_FAILURE;
     }
 
+    if (has_arg(argc, argv, "log-attributes")) {
+        sentry_value_t attributes = sentry_value_new_object();
+        sentry_value_t attr = sentry_value_new_attribute(
+            sentry_value_new_string("my_attribute"), NULL);
+        sentry_value_t attr_2 = sentry_value_new_attribute(
+            sentry_value_new_int64(INT64_MAX), "fermions");
+        sentry_value_t attr_3 = sentry_value_new_attribute(
+            sentry_value_new_int64(INT64_MIN), "bosons");
+        sentry_value_set_by_key(attributes, "my.custom.attribute", attr);
+        sentry_value_set_by_key(attributes, "number.first", attr_2);
+        sentry_value_set_by_key(attributes, "number.second", attr_3);
+        // testing multiple attributes
+        sentry_log_debug("logging with %d custom attributes", attributes, 3);
+        // testing no attributes
+        sentry_log_debug("logging with %s custom attributes",
+            sentry_value_new_object(), "no");
+        // testing overwriting default attributes
+        sentry_value_t param_attributes = sentry_value_new_object();
+        sentry_value_t param_attr = sentry_value_new_attribute(
+            sentry_value_new_string("parameter"), NULL);
+        sentry_value_t param_attr_2 = sentry_value_new_attribute(
+            sentry_value_new_string("custom-sdk-name"), NULL);
+        // TODO only sentry.message.parameter.X gets picked up by product
+        //  this will be in the envelope but not searchable!
+        sentry_value_set_by_key(
+            param_attributes, "message.parameter.0", param_attr);
+        sentry_value_set_by_key(
+            param_attributes, "sentry.sdk.name", param_attr_2);
+        // TODO discuss; param_attributes[message.parameter.0] gets overwritten
+        //  by 'and format string'. Is this expected?
+        sentry_log_fatal("logging with a custom parameter %s attributes",
+            param_attributes, "and format-string");
```

Comment:
IMO we can overwrite all default attributes except for string format values that are passed in; otherwise there would be a mismatch between the string template + `sentry.message.parameter.X` and the log `body` value